homeassistant/components/repetier/sensor.py

```diff
@@ -0,0 +1,294 @@
+            self._attributes = {'started': self._starttime,
+                                'seconds': self._secs}
+
+        return self._attributes
+
+    @Throttle(SCAN_INTERVAL)
+    def update(self):
+        """Update the sensor."""
+        import pyrepetier
+        import time
+        server = pyrepetier.Repetier(url=self._server['url'],
+                                     port=self._server['port'],
+                                     apikey=self._server['apikey'])
+        server.getPrinters()
+        if self._type == 'nozzle':
+            nozzle = server.Nozzle(self._id)
+            _LOGGER.debug(nozzle)
+            if nozzle is not None:
+                _LOGGER.debug("Nozzle %s Setpoint: %s, Temp: %s",
+                              self._nozzle, nozzle[self._nozzle]['tempset'],
+                              nozzle[self._nozzle]['temp'])
+                self._setpoint = nozzle[self._nozzle]['tempset']
+                self._state = nozzle[self._nozzle]['temp']
+            else:
+                _LOGGER.debug("Nozzle %s not found - is printer offline?",
+                              self._nozzle)
+                self._setpoint = None
+                self._state = None
+        elif self._type == 'bed':
+            bed = server.Bed(self._id)
+            if bed is not None:
+                _LOGGER.debug("Bed %s Setpoint: %s, Temp: %s",
+                              self._bed, bed[self._bed]['tempset'],
+                              bed[self._bed]['temp'])
+                self._setpoint = bed[self._bed]['tempset']
+                self._state = bed[self._bed]['temp']
+            else:
+                _LOGGER.debug("Bed %s not found - is printer offline?",
+                              self._bed)
+                self._setpoint = None
+                self._state = None
+        elif self._type == 'state':
+            state = server.State(self._id)
+            _LOGGER.debug("Printer %s current state: %s", self._name, state)
+            self._state = state
+        elif self._type == 'percentage':
+            state = server.Percent(self._id)
+            if state is not None:
+                self._state = round(state, 1)
+                self._jobname = server.JobName(self._id)
+                self._jobid = server.JobID(self._id)
+                self._totallines = server.TotalLines(self._id)
+                self._linessent = server.LinesSent(self._id)
+        elif self._type == 'remaining':
+            start = server.TimeStart(self._id)
+            end = server.PrintLength(self._id)
+            printtime = server.PrintTime(self._id)
+            if start is not None and end is not None and printtime is not None:
+                remaining = end - printtime
+                tend = start + round(end, 0)
+                dtime = datetime.utcfromtimestamp(tend).strftime(TIME_FORMAT)
+                self._secs = int(round(remaining, 0))
+                self._state = time.strftime('%H:%M:%S',
```

Comment:
But this is not a time stamp - it's a calculation in hours, minutes and seconds until end.
And the one below is the same, just for time elapsed since start.

Attributes for end timestamp and start timestamp will be changed to conform with ISO formatting in next push.

Reply:
Yes. We don't allow relative times in the state machine. The timestamp sensor class is an alternative for those cases. The frontend will allow the user to set a relative time representation of the absolute time point we set as state. So computer the absolute time stamp by adding current time to the relative time we have here.